examples/stock/company.v.py

```diff
@@ -1,21 +1,23 @@
 # Own shares of a company!
 company: address
-avail_shares: num
-total_shares: num
-price: wei_value
+total_shares: currency_value
+price: num (wei / currency)
 
```

Comment:
I checked it out and both `@constant` and `public` functions cost gas (every evm call does).  The issue is that the gas should only be taken from the sender when the state is changed.  Simply calling the blockchain costs gas but is free because it can be done locally (accessing 1 node instead of all of them)